web/docs/understand/operators/transformations/inspect.md

```diff
@@ -0,0 +1,59 @@
+# inspect
+
+Replaces the input with incremental metrics describing about the input.
+
+## Synopsis
+
+```
+inspect [--real-time]
+```
+
+## Description
+
+The `inspect` operator replaces the events or bytes it receives with incremental
+metrics describing the input using the following schema, respectively:
+
+```title="Events Metrics"
+type vast.metrics.events = record  {
+  timestamp: time,
+  schema: string,
+  schema_id: string,
+  events: uint64,
+}
+```
+
+```title="Bytes Metrics"
+type vast.metrics.bytes = record  {
+  timestamp: time,
+  bytes: uint64,
+}
+```
+
+### `--real-time`
+
+Emit metrics immediately rather than buffering at the cost of performance.
```

Comment:
As a user, I don't understand what I'm "paying" for. Can you describe details about the buffering, e.g., in terms of space or time.

Reply:
Suggestion for the description:

> Emit metrics immediately with every batch, rather than buffering until the upstream operator stalls, i.e., is idle or waiting for further input.
>
> The `--real-time` option is useful when `inspect` should emit data without latency.

Writing this, I realized that I would probably want `--real-time` to be the default. Have you ever encountered an issue with this in practice?

Reply:
I had a test setup with ~300M results and ran roughly the following with hardcoded timing code around the pipeline:

```
vast export json 'inspect | summarize sum(bytes) by schema'
```

With `--real-time`, the pipeline took over 100x longer because we sent batches of size 1 into the `summarize` operators.

I would argue that it's a very dangerous default, as you usually want to put the metrics into the pipeline because you want to work with them further.

Reply:
I see. If most of the workloads are of this kind, then it indeed is safer to opt out of batching.